### PR TITLE
Update of samples for UWP and current stable Xamarin.Forms

### PR DESCRIPTION
**Pull request by ellisnet**
Summary of proposed changes:
- Added UWP projects for any samples where UWP implementation did not exist.
- Implemented TextToSpeech for UWP for UsingDependencyService sample.
- Marked WinPhone projects (where they existed) as "Deprecated"
- Updated Prism NuGet packages in all projects (except WinPhone) to current stable: 6.3.0
- Updated Xamarin.Forms NuGet packages in all projects to current stable: 2.3.3.193
- Added description for ContosoCookbook sample to README file.
- Verified that all projects build (including WinPhone) and run (excluding WinPhone) with Visual Studio 2017.

### CODE REVIEW

File: ContosoCookbook/ContosoCookbook/ContosoCookbook.UWP/ContosoCookbook.UWP.nuget.targets [hunk 1]
```diff
@@ -5,7 +5,7 @@
     <RestoreTool Condition=" '$(RestoreTool)' == '' ">NuGet</RestoreTool>
     <ProjectAssetsFile Condition=" '$(ProjectAssetsFile)' == '' ">C:\GitHome\Prism-Samples-Forms\ContosoCookbook\ContosoCookbook\ContosoCookbook.UWP\project.lock.json</ProjectAssetsFile>
     <NuGetPackageRoot Condition=" '$(NuGetPackageRoot)' == '' ">$(UserProfile)\.nuget\packages\</NuGetPackageRoot>
-    <NuGetPackageFolders Condition=" '$(NuGetPackageFolders)' == '' ">$(UserProfile)\.nuget\packages\</NuGetPackageFolders>
+    <NuGetPackageFolders Condition=" '$(NuGetPackageFolders)' == '' ">C:\Users\jerem\.nuget\packages\</NuGetPackageFolders>
```
Comment (brianlagunas) [inline ContosoCookbook/ContosoCookbook/ContosoCookbook.UWP/ContosoCookbook.UWP.nuget.targets]: I see this in a few places.  This isn't going to work :)

**Comment by ellisnet** [comment]
UWP .targets files are fixed.  Thank you for catching that.

File: ContosoCookbook/ContosoCookbook/ContosoCookbook.UWP/ContosoCookbook.UWP.nuget.targets (hunk 1 above)
Comment (ellisnet) [inline ContosoCookbook/ContosoCookbook/ContosoCookbook.UWP/ContosoCookbook.UWP.nuget.targets]: Should be fixed with latest commit. Thank you for catching that.

**Comment by brianlagunas** [comment]
Thanks for the PR!

**Pull request merged**